keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/artifacts.kt

```diff
@@ -17,46 +17,26 @@
  */
 package com.netflix.spinnaker.keel.api
 
-import com.fasterxml.jackson.annotation.JsonIgnore
-import com.fasterxml.jackson.databind.JsonDeserializer
-import com.fasterxml.jackson.databind.annotation.JsonDeserialize
 import com.netflix.frigga.ami.AppVersion
 import com.netflix.rocket.semver.shaded.DebianVersionComparator
 import com.netflix.spinnaker.keel.api.SortType.INCREASING
 import com.netflix.spinnaker.keel.api.SortType.SEMVER
 import com.netflix.spinnaker.keel.exceptions.InvalidRegexException
-import com.netflix.spinnaker.keel.serialization.PropertyNamePolymorphicDeserializer
 import net.swiftzer.semver.SemVer
 import org.slf4j.LoggerFactory
 import org.springframework.util.comparator.NullSafeComparator
 
-/**
- * Strategy for how to sort versions of artifacts.
- */
-@JsonDeserialize(using = VersioningStrategyDeserializer::class)
-sealed class VersioningStrategy(
-  @JsonIgnore
-  open val comparator: Comparator<String>
-)
-
-@JsonDeserialize(using = JsonDeserializer.None::class)
-object DebianSemVerVersioningStrategy : VersioningStrategy(DEBIAN_VERSION_COMPARATOR) {
-  override fun toString(): String =
-    javaClass.simpleName
-
-  override fun equals(other: Any?): Boolean {
-    return other is DebianSemVerVersioningStrategy
-  }
-}
+val DockerArtifact.organization
+  get() = name.split("/").first()
 
-@JsonDeserialize(using = JsonDeserializer.None::class)
-data class DockerVersioningStrategy(
-  val strategy: TagVersionStrategy,
-  val captureGroupRegex: String? = null
-) : VersioningStrategy(TagComparator(strategy, captureGroupRegex)) {
-  override fun toString(): String =
-    "${javaClass.simpleName}[strategy=$strategy, captureGroupRegex=$captureGroupRegex]}"
-}
+val DockerArtifact.image
+  get() = name.split("/").last()
+
+val VersioningStrategy.comparator: Comparator<String>
+  get() = when (this) {
+    is DebianSemVerVersioningStrategy -> DEBIAN_VERSION_COMPARATOR
+    is DockerVersioningStrategy -> TagComparator(strategy, captureGroupRegex)
+  }
```

Comment:
The implementations of the comparators for `VersioningStrategy` rely on some 3rd party libs so it made sense to make this an extension and just move the model portion to `keel-api`. Also means it no longer requires `@JsonIgnore`.